packages/astro/src/assets/build/remote.ts

```diff
@@ -0,0 +1,55 @@
+import CachePolicy from 'http-cache-semantics';
+
+export type RemoteCacheEntry = { data: string; expires: number };
+
+export async function loadRemoteImage(src: string) {
+	try {
+		if (src.startsWith('//')) {
+			src = `https:${src}`;
+		}
+
+		const req = new Request(src);
+		const res = await fetch(req);
+
+		if (!res.ok) {
+			return undefined;
```

Comment:
It would be nice if we could log a warning/debug that says that the image returned a non-200 status code.